sdk/python/lib/pulumi_policy/policy.py

```diff
@@ -631,11 +643,22 @@ def AnalyzeStack(self, request, context):
                 resources.append(i.resource)
             args = StackValidationArgs(resources)
 
-            result = policy.validate(args, report_violation)
-            if isawaitable(result):
-                loop = asyncio.new_event_loop()
-                loop.run_until_complete(result)
-                loop.close()
+            try:
+                result = policy.validate(args, report_violation)
+                if isawaitable(result):
+                    loop = asyncio.new_event_loop()
+                    loop.run_until_complete(result)
+                    loop.close()
+            except UnknownValueError as e:
+                diagnostics.append(proto.AnalyzeDiagnostic(
+                    policyName=policy.name,
+                    policyPackName=self.__policy_pack_name,
+                    policyPackVersion=self.__policy_pack_version,
+                    message=f"can't run policy '{policy.name}' during preview: {e.message}",
```

Comment:
What does this look like when printed? Wondering if this should be more descriptive?

Reply:
I wondered the same actually.  Note that this is the same as TS though:

https://github.com/pulumi/pulumi-policy/blob/5f41c47c0945414d142e149745a88cd6b9334f64/sdk/nodejs/policy/server.ts#L269